src/Chapter4.hs

```diff
@@ -673,6 +722,29 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data Tree a = Leaf a
+            | Node (Tree a) a (Tree a) deriving (Show, Eq)
```

Comment:
That's one possible implementation of `Tree` and it's valid 👍🏻 But it would be more convenient to not have any fields in the `Leaf` constructor, so you can represent empty trees and generally define more flexible structures
```suggestion
data Tree a = Leaf
            | Node (Tree a) a (Tree a) deriving (Show, Eq)
```